src/render.rs

```diff
@@ -35,28 +35,34 @@ impl ForLoop {
 
 #[derive(Debug)]
 pub struct Renderer {
-    output: String,
     context: Json,
     ast: Node,
-    for_loops: Vec<ForLoop>
+    for_loops: Vec<ForLoop>,
+    render_error: Option<RenderError>
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct RenderError {
```

Comment:
can you put the errors in a separate file and maybe use https://github.com/tailhook/quick-error or structure them like https://github.com/kbknapp/clap-rs/blob/master/src/errors.rs ?